### ISSUE TITLE
complete_dictionary_with_modules doesn't add empty arrays for missing Tabxxxx

### ISSUE THREAD

**Issue by Daviid-P** [Dolibarr/dolibarr#22270]
### Bug

On `/core/lib/admin.lib.php` on the `complete_dictionary_with_modules()` function, when adding a module's own `$this->dictionaries` to `$taborder`,`$tabname`,`$tablib`,`$tabsql`,`$tabsqlsort`,`$tabfield`,`$tabfieldvalue`,`$tabfieldinsert`,`$tabrowid`,`$tabcond`,`$tabhelp`,`$tabcomplete`, if a module doesn't implement a `tabxxxx` in their `$this->dictionaries` array it will end up with a mismatched length.

I think this is because when adding the value it's done like `$tabxxxxx[] = $val;`

So if I make 2 modules, one that has this dictionaries definition:

```php
$this->dictionaries=array(
    'langs'=>'module1@module1',
    'tabname'=>array(MAIN_DB_PREFIX."c_module1_dict"),
    'tablib'=>array(""),
    'tabsql'=>array(''),
    'tabsqlsort'=>array(""),
    'tabfield'=>array(""),
    'tabfieldvalue'=>array(""),
    'tabfieldinsert'=>array(""),
    'tabrowid'=>array("rowid"),
    'tabcond'=>array($conf->module1->enabled),
    //'tabhelp'=>array(), // Module 1 does not have any tabhelp
);
```

And a second module (loaded after the first one) with this definition:

```php
$this->dictionaries=array(
    'langs'=>'module2@module2',
    'tabname'=>array(MAIN_DB_PREFIX."c_module2_dict"),
    'tablib'=>array(""),
    'tabsql'=>array(''),
    'tabsqlsort'=>array(""),
    'tabfield'=>array(""),
    'tabfieldvalue'=>array(""),
    'tabfieldinsert'=>array(""),
    'tabrowid'=>array("rowid"),
    'tabcond'=>array($conf->module2->enabled),
    'tabhelp'=>array(
        array('code'=>$langs->trans("EnterAnyCode").'<strong>Max 5 caracteres</strong>'),
    ),
);
```

We would end up with the `tabhelp` of the second module in the idx of the first module's dictionaries.

So when I go to `/admin/dict.php?id=idx_of_first_module` I would see the tabhelp of the second module.

### Expected and actual behavior

At the start of the function get the latest index of the arrays that are already predefined on `dict.php` and save it as `$last_index`

Then while adding each module's dictionaries add them like `$tabxxxxx[$latest_index + $i] = $val;`

`$i` being defined as a "sequencer of modules found" at the start of the `complete_dictionary_with_modules` function.

**Comment by Daviid-P** [comment]
I realize my fix won't work if a module defines 2 or more dictionaries, because `$last_index + $1` would be the same for  all dictionaries of that module, so we would end up with just the last dictionary defined of the module.

**Comment by Daviid-P** [comment]
I see this was somewhat fixed on #21530, but wouldn't it be easier to fix the way `complete_dictionary_with_modules` adds dictionaries to the `tabxxxx` arrays?

Something like:

1.  Initialize the `mod_tabfield`, `mod_tabfieldvalue`, `mod_tabfieldinsert` etc... arrays for the module as empty arrays.
2. Then loop through the module's `$this->dictionaries` and substitute the empty arrays for the actual values.
3. At the end we can do `$tabfield[] = $mod_tabfield;` while being sure that the amount will always be the same

Since a single module can define multiple dictionaries it might be better to use `array_merge` instead of `array[] = `

If that's not a good idea, then I think there's still a problem with only checking these 4 tabs if all of them should have the same length 
```php
$nbtabname != $nbtablib || $nbtablib != $nbtabsql || $nbtabsql != $nbtabsqlsort
```

I also see that now ModuleBuilder has a comment on the example dictionaries array:
```php
// Tooltip for every fields of dictionaries: DO NOT PUT AN EMPTY ARRAY
'tabhelp'=>array(array('code'=>$langs->trans('CodeTooltipHelp'), 'field2' => 'field2tooltip'), array('code'=>$langs->trans('CodeTooltipHelp'), 'field2' => 'field2tooltip'), ...),
```
But it's still possible that someone might think "Well I don't need `tabhelp` so I won't put an empty array, I'll just delete the whole thing"

We could iterate all `$objMod->dictionaries` arrays and check they have the same length:

```php
// First check all tabxxxx are defined
if (count($objMod->dictionaries) != $however_many_there_should_be) {
    // Error, not all tabs defined
}
// Then check all tabs have the same length
$count = count($objMod->dictionaries['tabname']);
foreach ($objMod->dictionaries as $tab_name => $tab_array) {
    if ($count != count($tab_array)) {
        // Error $tab_name has length count($tab_array) and it whould be $count, please define a $tab_name for every dictionary table even if empty
    }
}
```

**Comment by Daviid-P** [comment]
MAybe someone change this issue from Bug to Feature Request
